dao/src/main/java/se/fortnox/reactivewizard/db/precheck/PrecheckModule.java

```diff
@@ -0,0 +1,38 @@
+package se.fortnox.reactivewizard.db.precheck;
+
+import com.google.inject.Binder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.fortnox.reactivewizard.binding.AutoBindModule;
+
+import javax.inject.Singleton;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+
+@Singleton
+public class PrecheckModule implements AutoBindModule {
+    private static Logger LOG = LoggerFactory.getLogger(PrecheckModule.class);
+
+    public PrecheckModule() {
+        try {
+            Method    method         = PrecheckModule.class.getMethod("configure", Binder.class);
+            Parameter firstParameter = method.getParameters()[0];
+            if (!firstParameter.isNamePresent()) {
+                LOG.error("The project was not compiled with the -parameters option which is required for DAO parameter resolving. " +
+                    "Please see https://github.com/FortnoxAB/reactive-wizard/wiki/CompilerParameters for more information.");
+                System.exit(1);
```

Comment:
Should this be done differently? I figured we would want to exit to make sure the service doesn't start and people miss the error, since it won't work properly anyway.

Reply:
Nice! Much needed! Any way this can be incorporated into the existing db module? Feels like it adds some costs to the already slow binding by acting as a separate no-op module.

Reply:
Something like this? :)